File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {   
public:
    int subarraySum(vector<int>& nums, int k) {
        int high = 0, sum = 0, count=0;
        unordered_map<int,int>h_map;

        while(high<nums.size()){
            sum+=nums[high];
            if(sum == k) count++;
            count+= h_map[sum-k];
            h_map[sum]++;
            high++;
        }
        return count;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py

class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ans = 0
        prefixSum = 0
        hashMap = defaultdict(int)
        for num in nums:
            prefixSum+=num
            if prefixSum == k:
                ans+=1
            hashMap[prefixSum]+=1
            ans+= hashMap[prefixSum-k]
        return ans
